erpnext/controllers/queries.py

```diff
@@ -420,6 +420,43 @@ def bom(doctype, txt, searchfield, start, page_len, filters):
 		})
 
 
+@frappe.whitelist()
+@frappe.validate_and_sanitize_search_inputs
+def workstation_query(doctype, txt, searchfield, start, page_len, filters):
+	fields = get_fields("Workstation", ["name"])
+
+	searchfields = frappe.get_meta("Workstation").get_search_fields()
+	searchfields = " or ".join([field + " like %(txt)s" for field in searchfields])
+
+	exists_cond = ""
+	operation = filters and filters.pop("operation", None)
+	if operation:
+		exists_cond = """ and exists(select wo.name from `tabWorkstation Operation` wo
+			WHERE wo.parent = `tabWorkstation`.name AND wo.operation = {0})""".format(frappe.db.escape(operation))
```

Comment:
should show workstations if workstation does not have any applicable operation set